airflow/utils/file.py

```diff
@@ -23,11 +23,15 @@
 import errno
 import os
 import shutil
+from contextlib import contextmanager
```

Comment:
why move this line of import? If we move, I think it should sort by alphabetical order

Reply:
That's exactly what happened. The Python convention is to group by stdlib, 3rd party libs, application libs and sort alphabetically within those groups. Since `contextlib` is in the stdlib, it was moved to the 1st group, and placed after `import shutil` because alphabetical order.

Reply:
I see, sorry, miss that before. in that case, could you add line in between `import shutil` and `from contextlib import contextmanager` to distinguish these two categories import.

Reply:
Why? Both `shutil` and `contextlib` are part of Python's stdlib:

```python
# "special" group
from __future__ import absolute_import
from __future__ import unicode_literals

# stdlib group
import errno
import os
import shutil
from contextlib import contextmanager
from tempfile import mkdtemp

# 3rd party libs group
from zope import deprecation

# application libs
```

Reply:
What are you using to sort the import lines? That is not how I've seen it done before

Reply:
PyCharm. Correct me if wrong, but I thought all import sorting tools done it like that. Is there some other standard for it?

Reply:
Flake8 doesn't enforce this sort order/pep8 doesn't define it, so it's up to the tooling. The other way I've seen it done is

```
from contextlib import contextmanager
import errno
import os
import shutil
from tempfile import mkdtemp
```

i.e. sorted based on package name

(Not saying that is better, just a way)

Reply:
https://pypi.org/project/flake8-isort/ might be worth looking at. Though it probably would be one of those PRs that touches 900 files.

Reply:
Oh, except that module is GPL'd. I don't know if that matters as it is just for development tooling, and not a required part of the release.

Reply:
agree with @ashb , the import is sorted based on module alphabetical order.  Initially I thought you said contextmanager is considered as 3rd party. If they all belong to stdlib, @ashb's way should be the way to change.

Reply:
Great! More consistency is fine by me, I don't care about the "how" too much.

isort is available as a separate library with MIT license too: https://pypi.org/project/isort. Ran it over Airflow and it changed 77 files...

I'm not really into licensing, quick Google tells me anything is fine with MIT, but you need to include a copy of the license with your project. Not sure if that's okay?

Reply:
there is an ongoing AIP(https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-6+Enforce+the+usage+of+a+code+formatter) which should provide a solution to import order issue.

Reply:
Wanted to point out this PR which orders imports with flake8-import-order plugin: https://github.com/apache/airflow/pull/4784.